src/main/main.ts

```diff
@@ -609,13 +609,13 @@ function registerAllIpcListeners() {
         searchEngine.execute(searchResultItem, privileged)
             .then(() => {
                 userInputHistoryManager.addItem(userInput);
-                if (searchResultItem.hideMainWindowAfterExecution && config.generalOptions.hideMainWindowAfterExecution) {
-                    hideMainWindow();
-                } else {
-                    updateMainWindowSize(0, config.appearanceOptions);
-                }
             })
             .catch((err) => logger.error(err));
+        if (searchResultItem.hideMainWindowAfterExecution && config.generalOptions.hideMainWindowAfterExecution) {
+            hideMainWindow();
+        } else {
+            updateMainWindowSize(0, config.appearanceOptions);
+        }
```

Comment:
I am not sure if this is right. We just hide the window before we even know if the execution was successful or not. I agree that there is not enough visualization on the UI when the execution is in progress, so the user doesn't know what's happening. What are your thoughts on this?

Reply:
well I think its better to hide the window right away, then if the execution wasn't successful we can either show ueli again with a result that tells the user what went wrong or create a new window in electron to do that.